src/main/java/pl/edu/zut/mad/schedule/SchedulePresenter.kt

```diff
@@ -0,0 +1,59 @@
+package pl.edu.zut.mad.schedule
+
+import com.ognev.kotlin.agendacalendarview.models.CalendarEvent
+import io.reactivex.Observable
+import pl.edu.zut.mad.schedule.data.ScheduleRepository
+import pl.edu.zut.mad.schedule.data.model.ui.EmptyDay
+import pl.edu.zut.mad.schedule.util.DatesProvider
+import pl.edu.zut.mad.schedule.util.ModelMapper
+import pl.edu.zut.mad.schedule.util.NetworkConnection
+import pl.edu.zut.mad.schedule.data.model.api.Day as DayDb
+import pl.edu.zut.mad.schedule.data.model.ui.Day as DayUi
+
+
+internal class SchedulePresenter(private val repository: ScheduleRepository, private val mapper: ModelMapper,
+                        private val view: ScheduleMvp.View, private val datesProvider: DatesProvider,
+                        private val user: User, private val connection: NetworkConnection)
+    : ScheduleMvp.Presenter {
+
+    override fun logout() {
+        user.delete()
+        repository.delete()
+        view.showLoginView(null)
```

Comment:
Why null?

Reply:
In refresh method I pass album number which is directly used for fetching new schedule. At the moment refreshing is temporary and will be refactored. However, it should be done by other view method :)